test/functional/autoread/focus_spec.lua

```diff
@@ -0,0 +1,64 @@
+local helpers = require('test.functional.helpers')(after_each)
+local thelpers = require('test.functional.terminal.helpers')
+local clear = helpers.clear
+local retry = helpers.retry
+local nvim_prog = helpers.nvim_prog
+local feed_command = helpers.feed_command
+local feed_data = thelpers.feed_data
+
+if helpers.pending_win32(pending) then return end
+
+describe('autoread TUI FocusGained/FocusLost', function()
+  local screen
+
+  before_each(function()
+    clear()
+    screen = thelpers.screen_setup(0, '["'..nvim_prog
+      ..'", "-u", "NONE", "-i", "NONE", "--cmd", "set noswapfile noshowcmd noruler"]')
+    feed_command('set autoread')
+    feed_data("\034\016")
+  end)
+
+  it('external file change', function()
+    local path = 'xtest-foo'
+    local expected_addition = [[
+    line 1
+    line 2
+    line 3
+    line 4
+    ]]
+
+    helpers.write_file(path, '')
+    feed_command('edit '..path)
+    retry(2, 3 * screen.timeout, function()
```

Comment:
Why do we need retry here?

Reply:
This is something that I saw in the [tui_spec.lua#L914](https://github.com/neovim/neovim/blob/master/test/functional/terminal/tui_spec.lua#L914) file and thought that it must be because the function `feed_data` might not always be successful. I remember reading that 0 can be returned  (via chansend) for the cases when no bytes are written. So I thought we retry to ensure that the specified key was received by the child nvim instance. 

Although in that case I think I should leave the expect calls inside the retry block. I shouldn't have moved them out.

Reply:
I removed the retries. It works without them :)